core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -2435,6 +2435,17 @@ protected void convertFrom(
                 tableSampleSpec.isRepeatable(),
                 tableSampleSpec.getRepeatableSeed());
         bb.setRoot(new Sample(cluster, bb.root(), params), false);
+      } else if (sampleSpec instanceof SqlSampleSpec.SqlTableSampleRowLimitSpec) {
+        SqlSampleSpec.SqlTableSampleRowLimitSpec tableSampleRowLimitSpec =
+            (SqlSampleSpec.SqlTableSampleRowLimitSpec) sampleSpec;
+        convertFrom(bb, operands.get(0));
+        RelOptSamplingParameters params =
+            new RelOptSamplingParameters(
```

Comment:
Instead of reusing the `RelOptSamplingParameters`, can you just make a new object that is for this specific purpose?